mbf_utility/src/navigation_utility.cpp

```diff
@@ -69,12 +71,31 @@ bool getRobotPose(const TF &tf,
   return success;
 }
 
+/**
+ * @brief Returns true, ifthe given Quaternion is normalized.
+ *
+ * @param _q The quaternion to check.
+ * @param _epsilon The epsilon (squared distance to 1).
+ */
+static bool isNormalized(const geometry_msgs::Quaternion& _q, double _epsilon)
```

Comment:
without this we would return also empty if the quaternion would be invalid (then tf2::getYaw would return nan)

Reply:
indirectly - if the quaternion is ill formed we would silently return "out of costmap bounds" instead of telling what's wrong

Reply:
Would you rather move this change into a dedicated PR?

Reply:
should be, but I'm ok with merging it here (I'm not a purist)